ehr/src/org/labkey/ehr/utils/TriggerScriptHelper.java

```diff
@@ -1215,30 +1237,106 @@ private String checkOtherDrawsQuantity(String id, Date date, String rowObjectId,
+        {
+            StringBuilder errorMsgBuilder = new StringBuilder();
+
+            //always report the most severe overage
+            if (!overages.isEmpty())
+            {
+                errorMsgBuilder.append("Blood volume of ");
+                errorMsgBuilder.append(rowQuantity);
+                errorMsgBuilder.append(" (");
+                errorMsgBuilder.append(overages.descendingSet().iterator().next());
+                errorMsgBuilder.append(" over ");
+                errorMsgBuilder.append(interval);
+                errorMsgBuilder.append(" days) exceeds the allowable volume of ");
+                errorMsgBuilder.append(maxAllowable);
+                errorMsgBuilder.append(" mL (weight: ");
+                errorMsgBuilder.append(weight);
+                errorMsgBuilder.append(" kg).\n");
+            }
+
+            //report all the ones that are close to limit
+            for (double closeValue : closeToThreshold.descendingSet())
+            {
+                errorMsgBuilder.append("Limit notice! Blood volume of ");
+                errorMsgBuilder.append(rowQuantity);
+                errorMsgBuilder.append(" (");
+                errorMsgBuilder.append(closeValue);
+                errorMsgBuilder.append(" over ");
+                errorMsgBuilder.append(interval);
+                errorMsgBuilder.append(" days) is within ");
+                errorMsgBuilder.append(getThresholdLimit());
+                errorMsgBuilder.append(" mL of the max allowable limit of ");
+                errorMsgBuilder.append(maxAllowable);
+                errorMsgBuilder.append(" mL (weight: ");
+                errorMsgBuilder.append(weight);
+                errorMsgBuilder.append(" kg).\n");
+            }
+
+            return errorMsgBuilder.toString();
+        }
+        else
         {
-            return "Blood volume of " + rowQuantity + " (" + overages.descendingSet().iterator().next() + " over " + interval + " days) exceeds the allowable volume of " + maxAllowable + " mL (weight: " + weight + " kg)";
+            return null; // No errors
         }
+    }
+
+    public double getThresholdLimit()
+    {
+        return _bloodLimitThreshold;
+
+    }
+
+    public void setThresholdLimit(double threshold)
+    {
+        _bloodLimitThreshold = threshold;
+
+    }
+
+    public boolean getThresholdCheck()
+    {
+        return _checkThreshold;
+
+    }
+
+    public void setThresholdCheck(boolean checkIt)
+    {
+        _checkThreshold = checkIt;
 
-        return null;
     }
 
     public static class BloodInfo implements Comparable<BloodInfo>
     {
         private String _objectId;
         private Date _date;
         private double _quantity;
+        private boolean _inTransaction;
 
         public BloodInfo() {}
 
-        public BloodInfo(String objectId, Date date, Double quantity)
+        public BloodInfo(String objectId, Date date, Double quantity, boolean isInTransaction)
```

Comment:
Same question as above. It is necessary to set this?

Reply:
Yes, added a comment why the isInTransaction is needed